### ISSUE TITLE
etcd-operator no longer works

### ISSUE THREAD

**Issue by AshtonKem** [coredns/helm#69]
etcd-operator was deprecated two years ago, and does not seem to still work with newer versions of k8s. Installing gives me this error:

```
 helm install stable/etcd-operator --generate-name
WARNING: This chart is deprecated
Error: INSTALLATION FAILED: failed to install CRD crds/etcd-cluster-crd.yaml: resource mapping not found for name: "etcdclusters.etcd.database.coreos.com" namespace: "" from "": no matches for kind "CustomResourceDefinition" in version "apiextensions.k8s.io/v1beta1"
ensure CRDs are installed first
```

**Comment by chrisohaver** [comment]
This is the coredns helm repo.

**Comment by AshtonKem** [comment]
Yes, but the instructions for this chart specify that it requires another chart that is not just broken, but deprecated. I can’t file a bug with the etcd-operator chart, it’s deprecated and archived 

Given that etcd-operator is not going to come back, I think this chart needs to be updated with new instructions/dependencies. 

—
Ashton Kemerling

**Comment by chrisohaver** [comment]
Ah.  I wasn't aware of the third bullet use case... 

> CoreDNS as an external dns provider for kubernetes federation. This is a sub case of 'external dns service' which uses etcd plugin for CoreDNS backend. **This deployment mode as a dependency on etcd-operator chart, which needs to be pre-installed.**

Is this a hard dependency on etcd-operator as implied?  Or does this just mean that the mode requires etcd?

IOW, should this instead say:

"This deployment mode requires an etcd service to be present at install time."

**Comment by chrisohaver** [comment]
I think the instruction intend to convey the following:

1. This 3rd mode requires an etcd cluster.
2. This chart does not install an ectd cluster.
3. An etcd cluster must therefor be installed by some other means.

**Comment by HagaiBarel** [comment]
@chrisohaver according to https://groups.google.com/g/kubernetes-sig-multicluster/c/aZv-YuafSko it seems that kubernetes federation is deprecated and no other project is slated to replace it.

Given that, should we just remove the reference to kubernetes federation from the chart use cases?
